src/ProjectTemplates/Web.ProjectTemplates/content/BlazorServerWeb-CSharp/Areas/Identity/RevalidatingAuthenticationStateProvider.cs

```diff
@@ -1,98 +1,59 @@
 using System;
 using System.Security.Claims;
-using System.Threading;
 using System.Threading.Tasks;
 using Microsoft.AspNetCore.Components;
+using Microsoft.AspNetCore.Components.Server;
 using Microsoft.AspNetCore.Identity;
 using Microsoft.Extensions.DependencyInjection;
 using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
 
 namespace BlazorServerWeb_CSharp.Areas.Identity
 {
-    /// <summary>
-    /// An <see cref="AuthenticationStateProvider"/> service that revalidates the
-    /// authentication state at regular intervals. If a signed-in user's security
-    /// stamp changes, this revalidation mechanism will sign the user out.
-    /// </summary>
-    /// <typeparam name="TUser">The type encapsulating a user.</typeparam>
-    public class RevalidatingAuthenticationStateProvider<TUser>
-        : AuthenticationStateProvider, IDisposable where TUser : class
+    public class RevalidatingIdentityAuthenticationStateProvider<TUser>
+        : RevalidatingServerAuthenticationStateProvider where TUser : class
     {
         private readonly static TimeSpan RevalidationInterval = TimeSpan.FromMinutes(30);
-
-        private readonly CancellationTokenSource _loopCancellationTokenSource = new CancellationTokenSource();
         private readonly IServiceScopeFactory _scopeFactory;
-        private readonly ILogger _logger;
-        private Task<AuthenticationState> _currentAuthenticationStateTask;
+        private readonly IdentityOptions _options;
 
-        public RevalidatingAuthenticationStateProvider(
+        public RevalidatingIdentityAuthenticationStateProvider(
+            ILoggerFactory loggerFactory,
             IServiceScopeFactory scopeFactory,
-            SignInManager<TUser> circuitScopeSignInManager,
-            ILogger<RevalidatingAuthenticationStateProvider<TUser>> logger)
+            IOptions<IdentityOptions> optionsAccessor)
```

Comment:
Would this ever really be null? Options is easy to use in tests `Options.Create()`

Reply:
Yeah IOptions should really never be null

Reply:
OK. I was copying the existing logic from `SignInManager`, but will simplify it as you suggest.